lib/ansible/modules/hostname.py

```diff
@@ -155,9 +155,7 @@ class GenericStrategy(object):
     def __init__(self, module):
         self.module = module
         self.changed = False
-        self.hostname_cmd = self.module.get_bin_path('hostnamectl', False)
-        if not self.hostname_cmd:
-            self.hostname_cmd = self.module.get_bin_path('hostname', True)
+        self.hostname_cmd = self.module.get_bin_path('hostname', True)
```

Comment:
This was kept here for backward compatbility - https://github.com/ansible/ansible/pull/59974/files#r313368633

Reply:
I'm missing something, then. I'm not doing anything to change the check that decides if we should use the systemd strategy (where hostnamectl makes sense).

Right now in `GenericStrategy`, we check if `hostnamectl` exists, then compare its output (with no flags) against the `name` parameter, which is clearly wrong since calling `hostnamectl` with no flags produces multi-line, formatted output that we don't manipulate at all. But it won't matter anyway, since we check if the box is systemd-based anyway and switch to that strategy if it is.

All I'm proposing here is only looking for hostnamectl where it actually makes sense to use it.

Reply:
Do you have a case in mind where this does the wrong thing?

Reply:
Looking through the code more carefully, I think this change is correct. We should only being using `hostnamectl` where we pass it flags to return _only_ the hostname. That is currently the `SystemdStrategy` class. Otherwise the value returned from `self.hostname_cmd` would be not at all what we expect.

I don't think having this conditional buys us anything, nor does it help with backward compatibility.